doc/en/user/source/styling/mbstyle/cookbook/lines.rst

```diff
@@ -0,0 +1,295 @@
+
+.. figure:: ../../sld/cookbook/images/line_simpleline.png
+
+   Simple line
+
+Code
+~~~~
+
+:download:`Download the "Simple line" MBStyle <artifacts/mbstyle_line_simpleline.json>`
+
+.. code-block:: json
+  :linenos:
+
+  {
+    "version": 8,
+    "name": "simple-line",
+    "layers":[
+      {
+        "id": "simple-line",
+        "type": "line",
+        "paint": {
+          "line-color": "#000000",
+          "line-width": "3"
+            }
+          }
+        ]
+    }
+
+Details
+~~~~~~~
+
+There is one layer style for this MBStyle, which is the simplest possible situation. Styling
+lines is accomplished in the paint (**lines 8-10**). **Line 9** specifies the color of the line to be
+black (``'#000000'``), while **line 10** specifies the width of the lines to be 3 pixels.
+
+
+Line with border
+----------------
+
+This example shows how to draw lines with borders (sometimes called "cased lines").
+In this case the lines are drawn with a 3 pixel blue center and a 1 pixel wide gray border.
+
+.. figure:: ../../sld/cookbook/images/line_linewithborder.png
+
+   Line with border
+
+Code
+~~~~
+
+:download:`Download the "Line with border" MBStyle <artifacts/mbstyle_line_borderedline.json>`
+
+.. code-block:: json
+  :linenos:
+
+  {
+    "version": 8,
+    "name": "simple-borderedline",
+    "layers":[
+      {
+            "id": "simple-borderedline",
+            "type": "line",
+            "layout": {
+              "line-cap": "round"
+            },
+            "paint": {
+              "line-color": "#333333",
+              "line-width": "5"
+            }
+          },
+          {
+            "id": "simple-line",
+            "type": "line",
+            "layout": {
+              "line-cap": "round"
+            },
+            "paint": {
+              "line-color": "#6699FF",
+              "line-width": "3"
+            }
+          }
+        ]
+    }
+
+
+Details
+~~~~~~~
+
+Lines in MBStyle have no notion of a "fill", only "stroke". Thus, unlike points or polygons, it is not possible to style the "edge" of the line geometry. It is, however, possible to achieve this effect by drawing each line twice: once with a certain width and again with a slightly smaller width. This gives the illusion of fill and stroke by obscuring the larger lines everywhere except along the edges of the smaller lines.
+
+Since every line is drawn twice, the order of the rendering is *very* important.
+GeoServer renders ``layers`` in the order that they are presented in the MBStyle.
+In this style, the gray border lines are drawn first via the first layer style, followed by the blue center lines in a second layer style. This ensures that the blue lines are not obscured by the gray lines, and also ensures proper rendering at intersections, so that the blue lines "connect".
+
+In this example, **lines 5-15** comprise the first feature style, which is the outer line (or "stroke").

```

Comment:
Use "layer" instead of "feature style" here. Feature style is an SLD concept that doesn't quite carry over to MBStyle